test/unit/strings_spec.lua

```diff
@@ -138,3 +138,23 @@ describe('vim_strchr()', function()
     eq(nil, vim_strchr('«\237\175\191\237\188\128»', 0x10FF00))
   end)
 end)
+
+describe('vim_strrchr()', function()
+  local vim_strrchr = function(s, c)
+    local str = to_cstr(s)
+    local res = strings.vim_strrchr(str, c)
+    if res == nil then
+      return nil
+    else
+      return res - str
+    end
+  end
+  itp('works', function()
+    eq(0, vim_strrchr('abc', ('a'):byte()))
+    eq(1, vim_strrchr('abc', ('b'):byte()))
+    eq(2, vim_strrchr('abc', ('c'):byte()))
+    eq(3, vim_strrchr('abca', ('a'):byte()))
+    eq(4, vim_strrchr('abcaad', ('a'):byte()))
+    eq(nil, vim_strrchr('abc', ('d'):byte()))
```

Comment:
Need to test a multibyte string, e.g.:

    eq(6, vim_strrchr('abc生a生', ('a'):byte()))

Reply:
The function is documented as not supporting multi-byte. I also added an assert to help with enforcing this restriction.

Reply:
The string can still be multi-byte. Because `0 < c < 0x80` is encoded in UTF-8 as single byte (highest bit `0`), it never matches any part of a multi-byte sequence with two or more bytes (highest bit of any byte is `1`).